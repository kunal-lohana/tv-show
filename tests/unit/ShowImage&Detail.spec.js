import { shallowMount, config, createLocalVue } from "@vue/test-utils";
import { imageData } from "./testingData";
import Vuetify from "vuetify";
import Vue from "vue";
import ShowImageDetail from "@/components/ShowImage&Detail/ShowImage&Detail.vue";

config.showDeprecationWarnings = false;
Vue.config.slient = false;

describe("Show Image and Detail Component ", () => {
  let localVue = createLocalVue();
  localVue.use(Vuetify);
  Vue.use(Vuetify);
  let vuetify;
  beforeEach(() => {
    vuetify = new Vuetify();
  });

  it("should match snapshot", () => {
    const wrapper = shallowMount(ShowImageDetail, {
      localVue,
      vuetify,
      propsData: {
        data: imageData
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("set prop. data", async () => {
    const wrapper = shallowMount(ShowImageDetail, {
      localVue,
      vuetify
    });
    await wrapper.setProps({
      data: imageData
    });
    expect(wrapper.vm.data.length).toBe(imageData.length);
  });

  it("check data prop. data", () => {
    const wrapper = shallowMount(ShowImageDetail, {
      localVue,
      vuetify,
      propsData: {
        data: imageData
      }
    });
    expect(wrapper.vm.data.length).toBe(imageData.length);
  });

  it("check image url ", () => {
    const wrapper = shallowMount(ShowImageDetail, {
      localVue,
      vuetify,
      propsData: {
        data: imageData
      }
    });
    const imageUrl = wrapper
      .find('[data-test="show-slide-image"]')
      .attributes("src");
    expect(imageUrl).toEqual(imageData[0].image);
  });
});
