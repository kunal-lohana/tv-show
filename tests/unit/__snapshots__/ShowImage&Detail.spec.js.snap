// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show Image and Detail Component  should match snapshot 1`] = `
<v-flex-stub tag="div" xs6="" sm4="" md3="" lg3="" class="pa-4">
  <v-card-stub loaderheight="4" tag="div" color="secondary" width="200" dark="true" app="">
    <v-tooltip-stub opendelay="0" closedelay="0" contentclass="" bottom="true" fixed="true" openonhover="true" maxwidth="auto" nudgebottom="0" nudgeleft="0" nudgeright="0" nudgetop="0" nudgewidth="0" tag="span"> <span></span></v-tooltip-stub>
  </v-card-stub>
</v-flex-stub>
`;
